File: fase_2/pasto.cpp

#include <bits/stdc++.h>
using namespace std;

int neighborX[4] = {0, 0, 1, -1};
int neighborY[4] = {1, -1, 0, 0};

int n, m;
int vis[1010][1010];
char pasto[1010][1010];
bool fora_pasto;
int v=0,k=0,V=0,K=0;

bool isValid (int y, int x) {
  if (y < 0) return false;
  if (x < 0) return false;
  if (y >= n) return false;
  if (x >= m) return false;
  if (pasto[y][x]=='x') {fora_pasto=true; return false ;}
  if (pasto[y][x] == '#') return false;

  return true;
}


void DFS (int y, int x) {
  if (vis[y][x]) return;
    
  vis[y][x] = 1;
  if (pasto[y][x]=='v') v+= 1;
  if (pasto[y][x]=='k') k+= 1;
  for (int i = 0 ; i < 4 ; i++) {
    int newX = x + neighborX[i];
    int newY = y + neighborY[i];
    if (isValid(newY, newX)) {
        DFS(newY, newX);
    }
  }
}

int main() {
    cin >> n >> m;


    for (int i = 1 ; i <= n ; i++) {
      for (int j = 1 ; j <= m ; j++) {
        cin >> pasto[i][j];
      }
    }


    for (int i=0; i<=n+1; i++){pasto[i][0]='x'; pasto[i][m+1]='x';}
    for (int i=0; i<=m+1; i++){pasto[0][i]='x'; pasto[n+1][i]='x';}

   
    
    
    for (int i = 1 ; i <= n ; i++) {
      for (int j = 1 ; j <= m ; j++) {
        v=0; k=0;fora_pasto=false;
        if (pasto[i][j] != '#' && !vis[i][j]) {
            DFS(i, j);
        }
        if (!fora_pasto){
          if(v>=k){
              V+= v;
          } else{
              K+=k;
          }
        }
      }
    }

    cout << K << " "<< V << endl;
}

File: fase_2/pasto.py
import sys

sys.setrecursionlimit(2*31-1)

n, m = map(int, input().split())

V = 0
K = 0
pasto = []
nx = [1,0,-1,0]
ny = [0,1,0,-1]
vis = [[False]*m for _ in range(n)]
for _ in range(m):
    x = list(input())
    pasto.append(x)
    
def dfs(y,x):
    global k, v
    vis[y][x]= True
    if pasto[y][x]=="k":
        k+= 1
    if pasto[y][x]=="v":
        v+= 1
    for l in range(4):
        ay = y + ny[l] 
        ax = x + nx[l] 
        if 0<=ax<m and 0<=ay<n and pasto[ay][ax]!="#":
            dfs(ay, ax)
            


for i in range(n):
    for j in range(m):
        v = 0
        k = 0
        if pasto[i][j]== "." and not vis[i][j]:
            dfs(i,j)

        if v>=k:
            V+= v
        else:
            K+= k
print(K, V)